nanofi/include/sitetosite/CRawSocketProtocol.h

```diff
@@ -0,0 +1,198 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __CSITE2SITE_CLIENT_PROTOCOL_H__
+#define __CSITE2SITE_CLIENT_PROTOCOL_H__
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+#include <uuid/uuid.h>
+
+#include "api/nanofi.h"
+#include "CSiteToSite.h"
+#include "CPeer.h"
+
+#include "uthash.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define DESCRIPTION_BUFFER_SIZE 2048
+
+struct CRawSiteToSiteClient;
+
+int readResponse(struct CRawSiteToSiteClient* client, RespondCode *code);
+
+int writeResponse(struct CRawSiteToSiteClient* client, RespondCode code, const char * message);
+
+int readRequestType(struct CRawSiteToSiteClient* client, RequestType *type);
+
+int writeRequestType(struct CRawSiteToSiteClient* client, RequestType type);
+
+void tearDown(struct CRawSiteToSiteClient* client);
+
+int initiateResourceNegotiation(struct CRawSiteToSiteClient* client);
+
+int initiateCodecResourceNegotiation(struct CRawSiteToSiteClient* client);
+
+int negotiateCodec(struct CRawSiteToSiteClient* client);
+
+int establish(struct CRawSiteToSiteClient* client);
+
+void addTransaction(struct CRawSiteToSiteClient * client, CTransaction * transaction);
+
+CTransaction * findTransaction(const struct CRawSiteToSiteClient * client, const char * id);
+
+void deleteTransaction(struct CRawSiteToSiteClient * client, const char * id);
+
+void clearTransactions(struct CRawSiteToSiteClient * client);
+
+int handShake(struct CRawSiteToSiteClient * client);
+
+int bootstrap(struct CRawSiteToSiteClient * client);
+
+int complete(struct CRawSiteToSiteClient * client, const char * transactionID);
+
+int confirm(struct CRawSiteToSiteClient * client, const char * transactionID);
+
+int transmitPayload(struct CRawSiteToSiteClient * client, const char * payload, const attribute_set * attributes);
+
+int16_t sendPacket(struct CRawSiteToSiteClient * client, const char * transactionID, CDataPacket *packet, flow_file_record * ff);
+
+CTransaction* createTransaction(struct CRawSiteToSiteClient * client, TransferDirection direction);
+
+static const char * getResourceName(const struct CRawSiteToSiteClient * c) {
+  return "SocketFlowFileProtocol";
+}
+
+static const char * getCodecResourceName(const struct CRawSiteToSiteClient * c) {
+  return "StandardFlowFileCodec";
+}
+
+static RespondCodeContext *getRespondCodeContext(RespondCode code) {
+  unsigned int i;
+  for ( i = 0; i < sizeof(respondCodeContext) / sizeof(RespondCodeContext); i++) {
+    if (respondCodeContext[i].code == code) {
+      return &respondCodeContext[i];
+    }
+  }
+  return NULL;
+}
+
+// RawSiteToSiteClient Class
+struct CRawSiteToSiteClient {
+  // Batch Count
+  uint64_t _batchCount;
```

Comment:
I know this was copied and changed to a struct, but can we change this to use the rest of the project's guidelines for variable naming?

Reply:
Done, made it consistent.